gix-object/tests/object/tree/iter.rs

```diff
@@ -52,3 +57,47 @@ fn everything() -> crate::Result {
     );
     Ok(())
 }
+
+#[test]
+fn lookup_entry_toplevel() -> crate::Result {
+    let odb = utils::tree_odb()?;
+    let root_tree_id = hex_to_id("ff7e7d2aecae1c3fb15054b289a4c58aa65b8646");
+
+    let mut buf = Vec::new();
+    let root_tree = odb.find_tree(&root_tree_id, &mut buf)?;
+
+    let mut buf = Vec::new();
+    let entry = root_tree.lookup_entry_by_path(&odb, &mut buf, "bin").unwrap().unwrap();
+
+    assert!(matches!(entry, Entry { .. }));
+    assert_eq!(entry.filename, "bin");
+
+    Ok(())
+}
+
+#[test]
+fn lookup_entry_nested_path() -> crate::Result {
+    let odb = utils::tree_odb()?;
+    let root_tree_id = hex_to_id("ff7e7d2aecae1c3fb15054b289a4c58aa65b8646");
+
+    let mut buf = Vec::new();
+    let root_tree = odb.find_tree(&root_tree_id, &mut buf)?;
+
+    let mut buf = Vec::new();
+    let entry = root_tree
+        .lookup_entry_by_path(&odb, &mut buf, "file/a")
+        .unwrap()
+        .unwrap();
+
+    assert!(matches!(entry, Entry { .. }));
```

Comment:
I would deduplicate the tests and compare the whole entry, showing that one can get trees and files. I'd also show what happens if the requested entry doesn't exist.